nengo_loihi/simulator.py

```diff
@@ -488,22 +488,23 @@ def run_steps(self, steps):
             self._make_run_steps()
         try:
             self._run_steps(steps)
-        except Exception:
-            if "loihi" in self.sims:
+        except Exception as e:
+            if "loihi" in self.sims and self.sims["loihi"].use_snips:
+                # Need to write to board, otherwise it will wait indefinitely
                 h2c = self.sims["loihi"].nengo_io_h2c
                 c2h = self.sims["loihi"].nengo_io_c2h
 
-                printed = False
+                print(traceback.format_exc())
+                print("\nAttempting to end simulation...")
+
                 for _ in range(steps):
                     h2c.write(h2c.numElements, [0] * h2c.numElements)
                     c2h.read(c2h.numElements)
-                    if not printed:
-                        print(traceback.format_exc())
-                        print("\nAttempting to end simulation...")
-                        printed = True
+                self.sims["loihi"].wait_for_completion()
                 self.sims["loihi"].n2board.nxDriver.stopExecution()
                 self.sims["loihi"].n2board.nxDriver.stopDriver()
-                self.sims["loihi"].wait_for_completion()
+            else:
+                raise e
```

Comment:
You can actually just do `raise` here without `e` and without the `as e` above and it'll do the same thing

Reply:
Oh, cool, that's good to know!